semver/parse_comparator.ts

```diff
@@ -1,26 +1,51 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
-import { parse } from "./parse.ts";
 import type { Operator, SemVerComparator } from "./types.ts";
-import { COMPARATOR_REGEXP } from "./_shared.ts";
+import { COMPARATOR_REGEXP, parseBuild, parsePrerelease } from "./_shared.ts";
 import { comparatorMin } from "./comparator_min.ts";
 import { comparatorMax } from "./comparator_max.ts";
 import { ANY, NONE } from "./constants.ts";
 
+type REGEXP_GROUPS = {
+  operator: Operator;
+  major: string;
+  minor: string;
+  patch: string;
+  prerelease: string;
+  buildmetadata: string;
+};
+
 /**
  * Parses a comparator string into a valid SemVerComparator.
  * @param comparator
  * @returns A valid SemVerComparator
  */
 export function parseComparator(comparator: string): SemVerComparator {
-  const m = comparator.match(COMPARATOR_REGEXP);
+  const match = comparator.match(COMPARATOR_REGEXP);
+  const groups = match?.groups;
+
+  if (!groups) return NONE;
 
-  if (!m) {
-    return NONE;
-  }
+  const {
+    operator = "",
+    major,
+    minor,
+    patch,
+    prerelease,
+    buildmetadata,
+  } = groups as REGEXP_GROUPS;
+
+  const semver = major
+    ? {
+      major: parseInt(major),
```

Comment:
Now the major version is not checked against `Number.MAX_SAFE_INTEGER`, which was checked in `parse`

Reply:
I've fixed it with a centralized `parseNumber` function that parses and checks the number.